// otplib/classes/OTPUtils.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _hexToInt2 = require('../utils/hexToInt');

var _hexToInt3 = _interopRequireDefault(_hexToInt2);

var _intToHex2 = require('../utils/intToHex');

var _intToHex3 = _interopRequireDefault(_intToHex2);

var _isSameToken2 = require('../utils/isSameToken');

var _isSameToken3 = _interopRequireDefault(_isSameToken2);

var _leftPad = require('../utils/leftPad');

var _leftPad2 = _interopRequireDefault(_leftPad);

var _removeSpaces2 = require('../utils/removeSpaces');

var _removeSpaces3 = _interopRequireDefault(_removeSpaces2);

var _secretKey = require('../utils/secretKey');

var _secretKey2 = _interopRequireDefault(_secretKey);

var _setsOf2 = require('../utils/setsOf');

var _setsOf3 = _interopRequireDefault(_setsOf2);

var _stringToHex2 = require('../utils/stringToHex');

var _stringToHex3 = _interopRequireDefault(_stringToHex2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
 * OTP Utilities
 *
 * @class OTPUtils
 * @since 3.0.0
 * @author Gerald Yeo
 * @license MIT
 */
var OTPUtils = function () {
  function OTPUtils() {
    _classCallCheck(this, OTPUtils);
  }

  _createClass(OTPUtils, null, [{
    key: 'hexToInt',


    /**
     * @see {@link module:utils/hexToInt}
     */
    value: function hexToInt() {
      return _hexToInt3.default.apply(undefined, arguments);
    }

    /**
     * @see {@link module:utils/intToHex}
     */

  }, {
    key: 'intToHex',
    value: function intToHex() {
      return _intToHex3.default.apply(undefined, arguments);
    }

    /**
     * @see {@link module:utils/isSameToken}
     */

  }, {
    key: 'isSameToken',
    value: function isSameToken() {
      return _isSameToken3.default.apply(undefined, arguments);
    }

    /**
     * @see {@link module:utils/leftPad}
     */

  }, {
    key: 'pad',
    value: function pad() {
      return _leftPad2.default.apply(undefined, arguments);
    }

    /**
     * @see {@link module:utils/removeSpaces}
     */

  }, {
    key: 'removeSpaces',
    value: function removeSpaces() {
      return _removeSpaces3.default.apply(undefined, arguments);
    }

    /**
     * @see {@link module:utils/secretKey}
     */

  }, {
    key: 'generateSecret',
    value: function generateSecret() {
      return _secretKey2.default.apply(undefined, arguments);
    }

    /**
     * @see {@link module:utils/setsOf}
     */

  }, {
    key: 'setsOf',
    value: function setsOf() {
      return _setsOf3.default.apply(undefined, arguments);
    }

    /**
     * @see {@link module:utils/stringToHex}
     */

  }, {
    key: 'stringToHex',
    value: function stringToHex() {
      return _stringToHex3.default.apply(undefined, arguments);
    }
  }]);

  return OTPUtils;
}();

exports.default = OTPUtils;
